packages/components/src/unit-control/index.tsx

```diff
@@ -178,10 +178,7 @@ function UnforwardedUnitControl(
 				: undefined;
 			const changeProps = { event, data };
 
-			onChangeProp?.(
-				`${ validParsedQuantity ?? '' }${ validParsedUnit }`,
-				changeProps
-			);
```

Comment:
These changes have been extracted to #40796, and can therefore be removed from this PR once #40796 is merged (and this PR is rebased on top of `trunk`).